Add pull request template (#11)
jupyterhub/.github: .github/pull_request_template.md

Thank you for contributing to this repository!

You can help us review your changes by answering these questions.
They are all optional, but the more information you provide the easier it will be for us to review your changes.
Everyone here is a volunteer, so please help us out if you can.

If you want to discuss anything Jupyter related, or to meet other users and developers, please say hello on https://discourse.jupyter.org/ .

### What does this PR do?
<!--
Please summarise this PR.
Please also edit the PR title so that it contains enough context to go into a changelog.
Use "Fixes #<NUM>" if this fixes an existing issue.
It may help to list each change with an explanation of why it's needed- remember that what seems obvious to you may not be obvious to a reviewer.
-->

> 

Reviewer: Maybe to provide extra context we could add something like:

```
Type of change:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
```

> ### Is this PR related to another issue, or is it part of a larger body of work?

Reviewer 2: ```suggestion
### Is this PR related to an issue, or is it part of a larger body of work?
```

This part here makes me think that this should be the place where one would link the issue associated with this PR.
If so, I was thinking that it might be useful to add a tip in this section description about using keywords like `closes / fixes #issueNo` to automatically close the issue.
What do you think?

<!--
Please feel free to provide as much context or links to external sites as you want!
-->

### Does this PR introduce a breaking change?
<!--
If so what changes might users need to make in their applications due to this PR?
-->

### How can this PR be tested?

> <!--
> If this is not fully covered by the automated tests please help us by showing us the steps needed to verify the changes work.
> -->

Reviewer: ```suggestion
<!--
If this is not fully covered by the automated tests please help us by describing the tests that you ran to verify your changes. Make sure to provide as much detail so that we can reproduce your tests.
-->
```

> Thank you for contributing to this repository!
> 
> You can help us review your changes by answering these questions.
> They are all optional, but the more information you provide the easier it will be for us to review your changes.
> Everyone here is a volunteer, so please help us out if you can.
> 
> If you want to discuss anything Jupyter related, or to meet other users and developers, please say hello on https://discourse.jupyter.org/ .

Reviewer 2: I think if we don't add this part between `<!-- and -->`, then it will be shown on every PR description.
The message is lovely (:heart:), so I don't think there's any issue if we leave it in the description either.

```suggestion
<!--
Thank you for contributing to this repository!

You can help us review your changes by answering these questions.
They are all optional, but the more information you provide the easier it will be for us to review your changes.
Everyone here is a volunteer, so please help us out if you can.

If you want to discuss anything Jupyter related, or to meet other users and developers, please say hello on https://discourse.jupyter.org/ .
-->
```

<!--
Thank you for contributing to this repository!

You can help us review your changes by answering these questions.
They are all optional, but the more information you provide the easier it will be for us to review your changes.
Everyone here is a volunteer, so please help us out if you can.

> If you want to discuss anything Jupyter related, or to meet other users and developers, please say hello on https://discourse.jupyter.org/ .

Reviewer 3: Should we add in this link as well? https://discourse.jupyter.org/t/introduce-yourself/17/

Author: Could do, though since we've got the welcome bot maybe we could add the introduce-yourself link there, perhaps with a stronger emphasis on "say hello", and just have the general link here?

Reviewer 3: I agree with this but would actually reverse the suggestion. This is a PR template so folks using it are probably comfortable/familiar enough to make a contribution - we would just like to get to know them better (hence "say hello"). Whereas the bot's job is to orient newcomers who may be more likely to open a support request (hence general link). What do you think?

Also I really am splitting hairs here, so feel free to pass :)

-->

### What does this PR do?
<!--
Please indicate the type of change made by this PR (tick one or more of the boxes) and summarise the PR.

> Please also edit the PR title so that it contains enough context to go into a changelog.

Reviewer 3: Can we give an example of what a good PR title candidate for a changelog looks/reads like?

Author: I'm running into the same problem I had with providing an example body:
https://github.com/jupyterhub/.github/pull/11#issuecomment-702375567

What might be nice is if the next time one of us opens a non-trivial PR we follow this template, then we could link to a couple of good examples here?

Reviewer 3: Absolutely fine!
